### ISSUE TITLE
FUT postions always return 0 by instrument.get_postions()

### ISSUE THREAD

**Issue by homeyi** [ranaroussi/qtpylib#9]
Ran,
 i using Version 1.4.9, FUT postions always return 0 by instrument.get_postions().
CASH can  normally get postions ,but change FUT (i only test SGX XINA50 constructs) always retrun 0.
i test some times,found not any problem,can you help check this,thanks !

**Comment by ranaroussi** [comment]
Are you using TWS? If so - do you see the order getting filled?

**Comment by homeyi** [comment]
yes i use TWS,i can see the order is completed.

```
    def on_bar(self, instrument):
        print("\n\n--- BAR ---")

        # buy if position = 0, sell if in position > 0
        if instrument.positions['position'] == 0:
            print(">> BUY")
            instrument.buy(10)
        else:
            print(">> EXIT (SELL)")
            instrument.exit()


```

the position return 0 when have completed buy action,and not continue buy  again in next time.

**Comment by ranaroussi** [comment]
Looks like there's no problem with he buy order... As long as you're in position, the algo should't try to buy again. In the **next bar event**, provided you're still in position - the algo should exit/flatten the position. Isn't that what's happening?

**Comment by homeyi** [comment]
String thing,First order next bar not buy and sell again,and return position 0,
If interrupt by ctrl +C , And run again,then buy once and next bar do nothing.

**Comment by ranaroussi** [comment]
Can you run the same code with a different Futures contract (ES/CL for example)? Do you experience the same behavior?

**Comment by homeyi** [comment]
Now,I run the same code with a TF Futures contracts(("TF", "FUT", "NYBOT", "USD", 201612),),Met a new problem,It only could  get quote data, but Tick and  Bar could not get it.
Only STK and CASH contracts(EUR/USD) can work.

**Comment by ranaroussi** [comment]
I finally figured out what the problem is! Some instruments does not receive `RTVOLUME` events from TWS and therefore only get quotes. This is similar to the problem you reported in issue #3 

I've fixed that in version 1.4.96 by creating a "synthetic tick" for all instruments that does not receive `RTVOLUME`. Please upgrade to the latest version and let me know if the problem is resolved.

**Comment by homeyi** [comment]
Now TF Futures contracts is working ok. but XINA50 still return position 0,Maybe some different in them contract info? I can send you my paper account by e-mail,if need be.

**Comment by ranaroussi** [comment]
Not sure why the XINA50 behaves differently. Please send me your paper account to ran [at] aroussi.com and I'll try to figure this out.

Thanks!

**Comment by homeyi** [comment]
account have sent by email.

**Comment by ranaroussi** [comment]
This bug was due to a bug with ezIBpy that casued malformed `contractString` for Asian Futures (and therefore wasn't able to find the position of the the XINA50 contract). Anyway - this was fixed in [ezIBpy 1.12.24](https://github.com/ranaroussi/ezibpy), which is included in QTPyLib 1.4.97. 

Thanks for your patience!

**Comment by homeyi** [comment]
It's working OK!
